docs/modules/ROOT/pages/variables-and-params.adoc

```diff
@@ -0,0 +1,384 @@
+WHERE this0[this0] = $param0
+```
+
+And the following parameters object:
+
+```js
+{
+    param0: "The Matrix"
+}
+```
+
+Parameters can also be reused, like variables:
+
+```javascript
+const movie = new Cypher.Node({ labels: ["Movie"] });
+const titleProp = movie.property(movie);
+const titleParam = new Cypher.Param("The Matrix");
+
+const query = new Cypher.Match(movie).where(Cypher.eq(titleProp, titleParam)).return(titleParam);
+
+const {cypher, params} = query.build();
+```
+
+_Cypher_
+```cypher
+MATCH (this0:Movie)
+WHERE this0[this0] = $param0
+RETURN this0, $param0
+```
+
+_Params_
+```js
+{
+    param0: "The Matrix"
+}
+```
+
+In these cases, Cypher Builder takes care of providing a name to the parameter, and correctly linking it to the params object. 
+
+Note that if two instances of `Param` are used, then 2 separate parameters will be returned, regardless of the value, for example:
+
+```javascript
+const movie = new Cypher.Node({ labels: ["Movie"] });
+const titleProp = movie.property(movie);
+const titleParam1 = new Cypher.Param("The Matrix");
+const titleParam2 = new Cypher.Param("The Matrix")
+
+const query = new Cypher.Match(movie).where(Cypher.eq(titleProp, titleParam1)).return(titleParam2);
+
+const {cypher, params} = query.build();
+```
+
+_Cypher_
+```cypher
+MATCH (this0:Movie)
+WHERE this0[this0] = $param0
+RETURN this0, $param0
+```
+
+_Params_
+```js
+{
+    param0: "The Matrix"
+}
+```
+
+
+== Literal
+Literal values can be defined with `Cypher.Literal`. Literals behave like Parameters, but it will inject the value provided directly into the Cypher, serializing it as needed.
+
+For instance:
+
+```javascript
+const movie = new Cypher.Node({ labels: ["Movie"] });
+const titleProp = movie.property(movie);
+const titleLiteral = new Cypher.Literal("The Matrix")
+
+const query = new Cypher.Match(movie).where(Cypher.eq(titleProp, titleParam)).return(titleParam);
+
+const {cypher, params} = query.build();
+```
+
+_Cypher_
+```cypher
+MATCH (this0:Movie)
+WHERE this0[this0] = "The Matrix"
+RETURN this0, "The Matrix"
+```
+
+_Params_
+```js
+{ }
+```
+
+Note how the value `The Matrix` is not injected directly, but correctly serialized to a string in Cypher. The following values are supported by `Literal`:
+
+* String: `Cypher.Literal("Hello")` -> `"Hello"`
+* Number: `Cypher.Literal(5)` -> `5`
+* Boolean: `Cypher.Literal(true)` -> `true`
+* Array: `Cypher.Literal([5, "Hello"])` -> `[5, "Hello"]`
+* Null: `Cypher.Literal(null)` -> `NULL`
+
+
+[NOTE]
+====
+Note that Literal does not escape the values and code injection is a risk. It is generally recommended to use `Cypher.Param` instead.
+====
+
+=== NULL
+
+As a shortcut for `new Cypher.Literal(null)`, the constant `Cypher.Null` is available. This will be translated to `NULL`.
+
+
+== Aliasing
+// Maybe aliasing can be moved to a how-to guide
+Variables are commonly used for aliasing in a `WITH` or `RETURN` statement. To do that, simply pass a tuple of the value and its alias.
```

Comment:
This opening chapter should be slightly reformulated in case we turn this section into a new page. I will leave it like that for now until we make a decision about this.